server/src/test/java/keywhiz/KeywhizServiceTest.java

```diff
@@ -7,8 +7,24 @@
 import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.assertNull;
 import static org.junit.Assert.assertSame;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
 
 public class KeywhizServiceTest {
+  @Test
+  public void doesNotValidateDatabaseIfDatabaseValidationDisabledInConfig() {
+    KeywhizConfig config = mock(KeywhizConfig.class);
+    when(config.shouldValidateDatabase()).thenReturn(false);
+
+    ServiceContext context = ServiceContext.create();
+    KeywhizService service = context.getService();
+    service.validateDatabase(config);
+
+    verify(config, never()).getDataSourceFactory();
+  }
```

Comment:
Ideally we'd have some sort of databaseValidator.validateDatabase() call that we could check for, but the way that the FlyWay stuff is instantiated doesn't lend itself well to that kind of check.